nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/truncate_path_controller.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__TRUNCATE_PATH_CONTROLLER_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__TRUNCATE_PATH_CONTROLLER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "nav_msgs/msg/path.hpp"
+
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class TruncatePath : public BT::DecoratorNode
```

Comment:
Is this really a "controller"?

Reply:
Do you think this belongs as a behavior tree node? I'm on the fence - but I could go with it as long as there's some documentation about the use / need for it (e.g. a tutorial showing how to do a following task)

Reply:
Not really. Maybe it is better to rename it as` _node`

Reply:
@naiveHobo what do you think? You're good at naming :-)

Reply:
I have just checked BehaviorTree.CPP library, and most decorator are named `*_node`. I think this name is fine. We could change in the future anyway